File: 0501 - 1000/601-700/605. Can Place Flowers/soln1.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        size = len(flowerbed)
        count = 0
        i = 0
        while i<size:
            if flowerbed[i] == 0:
                if (i==0 or flowerbed[i-1]==0) and (i==size-1 or flowerbed[i+1]==0):
                    count += 1
                    flowerbed[i] = 1
            i = i+1
        return count >= n

File: 3001 - 3500/3201-3300/3226. Number of Bit Changes to Make Two Integers Equal/soln1.cpp
class Solution {
public:
    int minChanges(int n, int k) {
        // If there are bits in k that are 1 but not in n, return -1
        if ((k & ~n) != 0) return -1;
        // Count the number of 1 bits in n that need to be changed to 0 to match k
        int changes = 0;
        // XOR to find the bits that differ
        int diff = n ^ k;
        while (diff) {
            // Check if the least significant bit is different
            if (diff & 1) changes++;
            diff >>= 1; // Shift right to check the next bit
        }
        return changes;
    }
};